src/ray/raylet/node_manager.cc

```diff
@@ -123,6 +123,20 @@ std::string WorkerOwnerString(std::shared_ptr<WorkerInterface> &worker) {
   return buffer.str();
 }
 
+void NodeManagerConfig::ComputeFinalLabels(const std::string &self_node_id) {
```

Comment:
Instead of creating a new `final_labels` field, we can just update `labels`.

We can name this function as `AddDefaultLabels`.

Or we can even just do it in main.cc

```
node_manager_config.labels = parse_node_labels(labels_json_str);
node_manager_config.labels[kLabelKeyNodeID] = self_node_id;

```